packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts

```diff
@@ -124,10 +124,27 @@ export const useNotifiSubscribe: ({
   const getFirstDiscordTargetFromId = (
     newData: ClientData | null,
   ): string | undefined => {
+    const confirmedDiscordTarget = newData?.discordTargets.find(
+      (discordTarget) => discordTarget.isConfirmed === true,
+    );
+
+    if (confirmedDiscordTarget?.id) {
+      setDiscordTargetData({
+        discriminator: confirmedDiscordTarget?.discriminator ?? '',
+        id: confirmedDiscordTarget?.id ?? '',
+        username: confirmedDiscordTarget?.username ?? '',
+        isConfirmed: true,
+      });
```

Comment:
in general it'd be nice not to have side-effects from a function called `get...` -- maybe we can change this a little bit

Reply:
ah, i see. let me think about that some

Reply:
ah, i think i did complicate things. let me change that up